lib/services/terraform.js

```diff
@@ -299,6 +299,101 @@ class Terraform {
         return this.executeTerraform(name, 'tf-destroy', tfDestroySpawn);
     }
 
+    async status({ cloud, keys }) {
+        this.spinner = ora('Inspecting Orbs node').start();
+
+        const { name } = cloud;
+        this.setTargetPath(name);
+
+        if (fs.existsSync(path.join(this.targetPath, 'terraform.tfvars'))) {
+            this.spinner.succeed();
+            this.spinner.start(`Inspecting Terraform project internal state (reconciliation)..`);
+            await this.terraformRefresh({ name });
+            this.spinner.succeed();
+
+            this.spinner.start(`Inspecting state after reconciliation..`);
+            const currentState = await this.terraformShow();
+
+            const { outputs } = currentState.values;
+
+            if ('main_vpc_id' in outputs &&
+                'manager_dns' in outputs &&
+                'manager_ip' in outputs &&
+                'block_storage' in outputs) {
+                if (outputs.main_vpc_id.value.length > 0 &&
+                    outputs.manager_dns.value.length > 0 &&
+                    outputs.manager_ip.value.length > 0 &&
+                    outputs.block_storage.value.length > 0) {
+
+                    this.spinner.succeed();
+                    this.spinner.start(`Looking for the manager EC2 instance..`);
+
+                    await this.checkManagerLivenessWithAWSCLI({
+                        profile: keys.aws.profile,
+                        region: cloud.region,
+                        ip: outputs.manager_ip.value,
+                        dns: outputs.manager_dns.value,
+                    });
+                }
+            } else {
+                this.spinner.fail('State outputs missing important key(s)');
+                return {
+                    ok: false,
+                    message: 'Internal Terraform state missing keys that indicate infrastructure correctness on AWS'
+                };
+            }
+
+        } else {
+            this.spinner.succeed();
+            return false;
+        }
+    }
+
+    async checkManagerLivenessWithAWSCLI({ ip, dns, profile, region }) {
+        let result = await exec(`aws ec2 describe-instances --region ${region} --profile ${profile}`);
```

Comment:
Please use AWS SDK

Reply:
The AWS SDK will take us a bit more time to develop and to touch. For the sake of simplicity I suggest we skip using the aws cli for this as it's just a surface check for the instance status. What do you think? We already specify that installing the aws cli is a prerequisite for a long time now..

Reply:
I'd rather use SDK. It does not require a lot of boilerplate in this case.

Reply:
Alright, I'll change the implementation